rust/ql/lib/codeql/rust/internal/TypeInference.qll

```diff
@@ -97,7 +97,11 @@ private module Input1 implements InputSig1<Location> {
         id = 2
         or
         kind = 1 and
-        id = idOfTypeParameterAstNode(tp0.(DynTraitTypeParameter).getTypeParam())
+        id =
+          idOfTypeParameterAstNode([
+              tp0.(DynTraitTypeParameter).getTypeParam().(AstNode),
```

Comment:
`.(AstNode)` appears to be redundant.

Reply:
This is to help QL figure out what the element type of the set is. It complains without the `.(AstNode)` :)

Reply:
That's surprising, but fair enough.